app/models/teacher.rb

```diff
@@ -45,4 +46,28 @@ def send_magic_link(*)
   def send_devise_notification(notification, *args)
     devise_mailer.send(notification, self, *args).deliver_later
   end
+
+  def should_send_reminder_email?
+    all_reference_requests =
+      ReferenceRequest
+        .joins(:work_history)
+        .where(work_histories: { application_form_id: application_form.id })
+        .where(received_at: nil, expired_at: nil)
+    @remindable_reference_requests =
+      all_reference_requests.select do |reference_request|
+        reference_request.should_send_reminder_email?(
+          (reference_request.expires_at.to_date - Time.zone.today).to_i,
+          reminder_emails.count,
+        )
```

Comment:
I think we should move this logic to the `remindable` concern, since it has access to all these things, and it might allow us to have a single method `should_send_reminder_email?` for both teachers and requestables.

For example:

```rb
def days_until_expired
  (reference_request.expires_at.to_date - Time.zone.today).to_i
end
```